### ISSUE TITLE
DatePicker: AutoOK prop?

### ISSUE THREAD

**Issue by alexamart** [react-toolbox/react-toolbox#229]
Hi, i'm loving react-toolbox :), just finding a little too much having to click the OK button on the DatePicker; having the selector closed after picking a day is very useful in most cases. 

Also while on dates, how about some options to support I18n?

**Comment by javivelasco** [comment]
Hi @alexamart! I'm very happy you like it :)

I think it definitively would give some flexibility. I'm going to add `onCalendarChange` prop to the picker so you can subscribe by passing a function that can retrieve the value and update the dialog setting `active` to false. The difference with `onChange` would be precisely that in the first one you have to update the value yourself while the second one happens once you accepted the dialog. I need to try it but I think it's an easy change that would be available via patch during the next couple of days.

About i18n I think it's very important but we still need to prepare the components to work well on mobile and add some layout components. Contributions with first steps with i18n are very welcome!

Thanks for your kind words!

**Comment by karlmacklin** [comment]
I'll second this comment, and also mention something I've thought about for this library and material-ui.com version (so maybe it's more of a Material UI thing in general):
A combined datepicker and timepicker would be very useful. For example, you want to set a reminder or a specific time on a specific day for something. To chain both of the 'pickers' in sequence, or something?
Just a thought!

**Comment by javivelasco** [comment]
I think it makes sense from a consumer point of view. All depends on whether we want to stick hard to the spec or if it's better to make it flexible with requirements made from people. I would happy to add stuff out of the spec and as a last priority :)

**Comment by KerenChandran** [comment]
@javivelasco The main issue is solved in 7dfa13fb7538732eaf36db6e6b3ffbf3105a4f04. As for the i18n support, we could open another ticket and close this one.

**Comment by javivelasco** [comment]
Yep

**Comment by joeblew99** [comment]
I would really like to see i18n followed up.
Whats the current state and thinking on this ?

I noticed that the team at material-ui have not fully done it. they ended up with ramming it n via react props, and so you end up with lots of prop.

**Comment by javivelasco** [comment]
Not sure how can it be implemented without messing too much. Honestly I did not spent time thinking on this yet..

**Comment by gedw99** [comment]
I know what you mean about messing too much. I noticed the other teams working on react had problems. Angular2 handled it from the start because its so disruptive to the code base, etc.

 i18n is really a cross cutting concern in project:
- front end needs it
- front end validation messages need it.
- backend needs it.

So, I have a weird idea, and would like you to think about it.
What if the translated text can from a separate service (like a web server) at runtime ? I have worked with these before. But for react-toolbox what you it mean ?
- Maybe each component that needs it can call for it at the level they are running and not need them to be passed in as props. 
- A standard storage (caching in reality) layer using local storage is included, so that whenever a component calls for the translation text, it will hit the local storage first, and if a cache miss occurs will get it from the server.

**Comment by javivelasco** [comment]
I know the project quite well because I'm working in an project using it. I propose to open an issue and:
- Identify components that should be internationalized.
- Think about use cases and provide a function to pass the appropriated values at rendering time. For example, localize a Date would be different from a String for obvious reasons. The former needs some transformation and the latter it's just a Text.

Probably turning i18n strings into functions that receive the appropiated values would be enough. Also, having functional components this is really easy because we are externalizing the state so anyone using it can bind the values to the functions generating localized strings.

**Comment by joeblew99** [comment]
@javivelasco 

I made a new issue here:
https://github.com/react-toolbox/react-toolbox/issues/623

I tried to highlight the goal, but also the architectural choices a bit.
Also the fact that it fits into a bigger picture, and is not easy to get right.

I think its every single component when i think about it realistically.  Maybe i am wrong.
SO its a framework level thing.

Lets try and work this out and move it forward in the new Issue :)
